lib/swoc/CMakeLists.txt

```diff
@@ -0,0 +1,132 @@
+        include/swoc/Vectray.h
+        )
+
+# These are external but required.
+set(EXTERNAL_HEADER_FILES
+        include/swoc/ext/HashFNV.h
+        )
+
+set(CC_FILES
+        src/bw_format.cc
+        src/bw_ip_format.cc
+        src/ArenaWriter.cc
+        src/Errata.cc
+        src/swoc_ip.cc
+        src/MemArena.cc
+        src/RBTree.cc
+        src/swoc_file.cc
+        src/TextView.cc
+        src/string_view_util.cc
+        )
+
+add_library(libswoc SHARED ${CC_FILES})
+set_target_properties(libswoc PROPERTIES OUTPUT_NAME swoc-${LIBSWOC_VERSION})
+if (CMAKE_COMPILER_IS_GNUCXX)
+    target_compile_options(libswoc PRIVATE -fPIC -Wall -Wextra -Werror -Wnon-virtual-dtor -Wpedantic)
+endif()
+
+add_library(libswoc-static STATIC ${CC_FILES})
+set_target_properties(libswoc-static PROPERTIES OUTPUT_NAME swoc-static-${LIBSWOC_VERSION})
+if (CMAKE_COMPILER_IS_GNUCXX)
+    target_compile_options(libswoc-static PRIVATE -fPIC -Wall -Wextra -Werror -Wnon-virtual-dtor -Wpedantic)
+endif()
+
+# Not quite sure how this works, but I think it generates one of two paths depending on the context.
+# That is, the generator functions return non-empty strings only in the corresponding context.
+target_include_directories(libswoc-static
+        PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+        $<INSTALL_INTERFACE:include>
+        )
+
+target_include_directories(libswoc
+        PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+        $<INSTALL_INTERFACE:include>
+        )
+
+if (LIBSWOC_INSTALL)
+    # These install target variables are created by GNUInstallDirs.
+    install(TARGETS libswoc-static
+            EXPORT libswoc-static-config
+            ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} # seems to have no effect.
+            )
+    install(TARGETS libswoc
+            EXPORT libswoc-config
+            ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} # seems to have no effect.
+            )
+    install(DIRECTORY include/swoc DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+
+    set(link-target libswoc.so)
+    set(link-src $<TARGET_FILE_NAME:libswoc>)
+    install(CODE
+            "execute_process(COMMAND bash -c \"echo Linking ${link-src} to ${link-target};cd ${CMAKE_INSTALL_FULL_LIBDIR} && (rm -f ${link-target};ln -s ${link-src} ${link-target})
+        \")"
+            )
+
+    install(EXPORT libswoc-static-config
+            NAMESPACE libswoc::
+            DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libswoc
+            )
+
+    install(EXPORT libswoc-config
+            NAMESPACE libswoc::
+            DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libswoc
+            )
+
+    #    set(PKG_CONFIG_FILE libswoc.pc)
+    #    set(PKG_STATIC_CONFIG_FILE libswoc-static.pc)
+    #    configure_file("libswoc.pc.cmake" ${PKG_CONFIG_FILE} @ONLY)
+    #
+    # configure_file("libswoc.pc.cmake" libswoc.pc @ONLY)
+    # configure_file("libswoc-static.pc.cmake" libswoc-static.pc @ONLY)
```

Comment:
I had to comment these `configure_file` lines out since the source files are missing.  Otherwise this file is the same as swoc upstream.

Reply:
Hmmm. I might to put those back in so that the internal libswoc can generate accurate pkg-config files. That would make it easier for external consumption to switch.